lib/Conversion/CalyxNative/CalyxNative.cpp

```diff
@@ -0,0 +1,150 @@
+//===- CalyxNative.cpp - Invoke the native Calyx compiler
+//----------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+//===----------------------------------------------------------------------===//
+//
+// This is the main Calyx to HW Conversion Pass Implementation.
+//
+//===----------------------------------------------------------------------===//
+
+#include "../PassDetail.h"
+
+#include "circt/Conversion/CalyxNative.h"
+#include "circt/Dialect/Calyx/CalyxEmitter.h"
+#include "circt/Dialect/SV/SVDialect.h"
+#include "circt/Dialect/Seq/SeqOps.h"
+#include "mlir/Parser/Parser.h"
+#include "mlir/Support/FileUtilities.h"
+#include "llvm/Support/MemoryBuffer.h"
+#include "llvm/Support/Path.h"
+#include "llvm/Support/Process.h"
+#include "llvm/Support/ToolOutputFile.h"
+
+using namespace mlir;
+using namespace circt;
+
+/// ConversionPatterns.
+
+/// Pass entrypoint.
+
+namespace {
+class CalyxNativePass : public CalyxNativeBase<CalyxNativePass> {
+public:
+  void runOnOperation() override;
+};
+} // end anonymous namespace
+
+void CalyxNativePass::runOnOperation() {
+  ModuleOp root = getOperation();
+
+  // User must specify location of the calyx primitive library.
+  if (primitiveLib == "") {
+    // XXX(rachitnigam): Probably a bad idea to hard code the name of the flag.
+    // Is there a better way?
+    root.emitError("primitive library not specified. Please specify it using "
+                   "--lower-using-calyx-native=\"primitives=<path>\"");
+    return;
+  }
+
+  SmallString<32> execName = llvm::sys::path::filename("calyx");
+  auto exeMb = llvm::sys::findProgramByName(execName);
```

Comment:
I made some similar comments below, but I think some of these autos should be given types, i.e., I wasn't sure if it returned a unique_ptr, an error code, etc.